subsys/llext/llext_load.c

```diff
@@ -574,7 +597,10 @@ static int llext_copy_symbols(struct llext_loader *ldr, struct llext *ext,
 				}
 			}
 
-			if (pre_located) {
+			const char *s_name = llext_string(ldr, ext, LLEXT_MEM_SHSTRTAB,
+							  shdr->sh_name);
+
+			if (pre_located && section_addr >= LLEXT_SECTION_MIN_ADDR) {
```

Comment:
This also feels specific - although maybe a per-section flag could work (I assume they are the sections _not_ ignored by the above check).